src/FSharp.Control.TaskSeq.Test/TaskSeq.SkipWhile.Tests.fs

```diff
@@ -0,0 +1,332 @@
+        | true, true -> fun pred -> TaskSeq.skipWhileInclusiveAsync (pred >> Task.fromResult)
+
+module EmptySeq =
+
+    // TaskSeq-skipWhile+A stands for:
+    // skipWhile + skipWhileAsync etc.
+
+    [<Theory; ClassData(typeof<TestEmptyVariants>)>]
+    let ``TaskSeq-skipWhile+A has no effect`` variant = task {
+        do!
+            Gen.getEmptyVariant variant
+            |> TaskSeq.skipWhile ((=) 12)
+            |> verifyEmpty
+
+        do!
+            Gen.getEmptyVariant variant
+            |> TaskSeq.skipWhileAsync ((=) 12 >> Task.fromResult)
+            |> verifyEmpty
+    }
+
+    [<Theory; ClassData(typeof<TestEmptyVariants>)>]
+    let ``TaskSeq-skipWhileInclusive+A has no effect`` variant = task {
+        do!
+            Gen.getEmptyVariant variant
+            |> TaskSeq.skipWhileInclusive ((=) 12)
+            |> verifyEmpty
+
+        do!
+            Gen.getEmptyVariant variant
+            |> TaskSeq.skipWhileInclusiveAsync ((=) 12 >> Task.fromResult)
+            |> verifyEmpty
+    }
+
+module Immutable =
+
+    // TaskSeq-skipWhile+A stands for:
+    // skipWhile + skipWhileAsync etc.
+
+    [<Theory; ClassData(typeof<TestImmTaskSeq>)>]
+    let ``TaskSeq-skipWhile+A filters correctly`` variant = task {
+        // truth table for f(x) = x < 5
+        // 1 2 3 4 5 6 7 8 9 10
+        // T T T T F F F F F F (stops at first F)
+        // x x x x _ _ _ _ _ _ (skips exclusive)
+        // A B C D E F G H I J
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhile ((>) 5) // skip while less than 5
+            |> verifyDigitsAsString "EFGHIJ"
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileAsync (fun x -> task { return x < 5 })
+            |> verifyDigitsAsString "EFGHIJ"
+    }
+
+    [<Theory; ClassData(typeof<TestImmTaskSeq>)>]
+    let ``TaskSeq-skipWhile+A does not skip first item when false`` variant = task {
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhile ((=) 0)
+            |> verifyDigitsAsString "ABCDEFGHIJ" // all 10 remain!
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileAsync ((=) 0 >> Task.fromResult)
+            |> verifyDigitsAsString "ABCDEFGHIJ" // all 10 remain!
+    }
+
+    [<Theory; ClassData(typeof<TestImmTaskSeq>)>]
+    let ``TaskSeq-skipWhileInclusive+A filters correctly`` variant = task {
+        // truth table for f(x) = x < 5
+        // 1 2 3 4 5 6 7 8 9 10
+        // T T T T F F F F F F (stops at first F)
+        // x x x x x _ _ _ _ _ (skips inclusively)
+        // A B C D E F G H I J
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusive ((>) 5)
+            |> verifyDigitsAsString "FGHIJ" // last 4
+
+        do!
+            Gen.getSeqImmutable variant
+            |> TaskSeq.skipWhileInclusiveAsync (fun x -> task { return x < 5 })
```

Comment:
need symmetry with preceding conditon to be able to scan this, i.e. make the last one `fun x -> x < 5`

Reply:
Fixed